### ISSUE TITLE
[FP] `Callable parameter usage violates definition` for object callables

### ISSUE THREAD

**Issue by jdreesen** [kalessil/phpinspectionsea#1777]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 3.0.12 |
| Language level | PHP 7.4                       |

Current behaviour
--

Inspection `Callable parameter usage violates definition` is triggered with message `[EA] It seems to be always false (no object types).` when checking if a `callable` parameter value is `instanceof` something.

Expected behaviour
--

Inspection doesn't get triggered.

Example
--

![grafik](https://user-images.githubusercontent.com/424602/142052415-92785bda-a016-45dc-b3eb-1f13670d0e41.png)

Will output:
```
new B
-> instance of A
-> instance of B

function
-> instance of Closure

string
```

See: https://3v4l.org/CVcVt

Environment details
---
PhpStorm 2021.2.3
Build #PS-212.5457.49, built on October 13, 2021
Runtime version: 11.0.12+7-b1504.40 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.4.0-90-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 4
Registry: debugger.watches.in.variables=false, run.processes.with.pty=TRUE
Current Desktop: ubuntu:GNOME

**Comment by ea-inspections-team** [comment]
Thank you for reporting @jdreesen! Fixed!
